spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -710,58 +692,46 @@ def restart_kernel(self):
         Licensed under the BSD license
         """
         sw = self.shellwidget
+        if not sw.kernel_manager:
+            sw._append_plain_text(
+                _('Cannot restart a kernel not started by Spyder\n'),
+                before_prompt=True
+            )
+            self._hide_loading_page()
+            return
 
-        if not running_under_pytest() and self.ask_before_restart:
-            message = _('Are you sure you want to restart the kernel?')
-            buttons = QMessageBox.Yes | QMessageBox.No
-            result = QMessageBox.question(self, _('Restart kernel?'),
```

Comment:
The messagebox is moved to the main_wodget, so that `restart_kernel` just restarts the kernel